exasol/slc/internal/tasks/test/run_db_generic_language_tests.py

```diff
@@ -46,12 +48,18 @@ def run_task(self):
 
     def run_test(
         self, language: str, test_folder: str
-    ) -> Generator[RunDBTestsInDirectory, Any, RunDBTestDirectoryResult]:
+    ) -> Generator[BaseTask, Any, RunDBTestDirectoryResult]:
+        #
+        # Correct return type is Generator[RunDBTestsInDirectory, Any, List[RunDBTestResult]]
+        # TODO: Fix after https://github.com/exasol/integration-test-docker-environment/issues/445
+        #
```

Comment:
MyPy complains here now if using:
```
Generator[RunDBTestsInDirectory, Any, RunDBTestDirectoryResult]
```
because in line 62
```
test_result_future = yield from self.run_dependencies(task)
```
`run_dependencies()` returns a`BaseTask`